### PR TITLE
Optimization/pre warm intra block cache fix nonce collisions

### PR DESCRIPTION
**Pull request by LukaszRozmej**
## Changes

- Should fix nonce collisions when pre-loading block

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a change that causes existing functionality not to work as expected)
- [x] Optimization
- [ ] Refactoring
- [ ] Documentation update
- [ ] Build-related changes
- [ ] Other: _Description_

#### Requires testing

- [ ] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [ ] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

**Review by benaadams — APPROVED** [review]

**Comment by LukaszRozmej** [comment]
Can you give feedback how is it performing now?

**Review by benaadams — COMMENTED** [review]
Would it not be better to batch the txs from same sender and run them sequentially?

**Comment by benaadams** [comment]
> Would it not be better to batch the txs from same sender and run them sequentially?

Something more like this? https://github.com/NethermindEth/nethermind/pull/7467

**Review by benaadams — APPROVED** [review]

File: src/Nethermind/Nethermind.Consensus/Processing/BlockCachePreWarmer.cs [hunk 1]
```diff
@@ -143,6 +144,20 @@ private void WarmupTransactions(ParallelOptions parallelOptions, IReleaseSpec sp
...
+                        Address senderAddress = tx.SenderAddress!;
+                        if (!scope.WorldState.AccountExists(senderAddress))
+                        {
+                            scope.WorldState.CreateAccountIfNotExists(senderAddress, UInt256.Zero, UInt256.One);
+                        }
+                        else
+                        {
+                            scope.WorldState.IncrementNonce(senderAddress);
+                        }
+                    }
```
Comment (benaadams) [inline src/Nethermind/Nethermind.Consensus/Processing/BlockCachePreWarmer.cs:159]:
```suggestion
                    Address senderAddress = tx.SenderAddress!;
                    for (int j = 0; j <= i; j++)
                    {
                        if (block.Transactions[j].Sender != senderAddress) continue;
                        
                        if (!scope.WorldState.AccountExists(senderAddress))
                        {
                            scope.WorldState.CreateAccountIfNotExists(senderAddress, UInt256.Zero, UInt256.One);
                        }
                        
                        scope.WorldState.IncrementNonce(senderAddress);
                    }
```

**Review by benaadams — APPROVED** [review]

**Review by benaadams — APPROVED** [review]

**Pull request merged**